src/raw_upload.rs

```diff
@@ -0,0 +1,101 @@
+use base64::prelude::*;
+use pyo3::prelude::*;
+use std::collections::HashSet;
+use std::io::prelude::*;
+
+use flate2::bufread::ZlibDecoder;
+
+use quick_xml::reader::Reader;
+use serde::Deserialize;
+
+use crate::junit::{get_position_info, use_reader};
+use crate::testrun::ParsingInfo;
+use crate::ParserError;
+
+#[derive(Deserialize, Debug, Clone)]
+struct TestResultFile {
+    filename: String,
+    #[serde(skip_deserializing)]
+    _format: String,
```

Comment:
I think you can just leave these off if they are not being used. Serde will just ignore unknown fields.